src/zenml/models/model_models.py

```diff
@@ -438,31 +430,23 @@ class ModelVersionArtifactRequestModel(
 class ModelVersionArtifactResponseModel(
     ModelVersionArtifactBaseModel, WorkspaceScopedResponseModel
 ):
-    """Model version link with artifact response model.
-
-    link_version: The version of the link (always 1 for not versioned links).
-    """
-
-    link_version: int
```

Comment:
How is versioning going to work inside a Model Version then? It is clear for versioned artifacts, but for ones with overwrite link?

Reply:
Essentially the `link_version` was simply moved to `artifact.version` and a model version can be linked to several artifact versions now.

Regarding `overwrite_link`, I'm actually not fully sure what that does in the first place, I'll get back to this question once we have docs for the model watch tower

Reply:
Or do you think we should keep the `overwrite` arg and change the functionality to make it unlink all artifacts with the same name?

Reply:
TBH I don't fully get the use case behind this arg yet so might need some clarification

Reply:
Non-versioned artifact links seem to me a bit of a strange concept anyway, if one creates a new version every run - the version is always one, otherwise, there is no harm in keeping it versioned in any case. 👍🏼